telegram/_bot.py

```diff
@@ -356,11 +356,19 @@ def __deepcopy__(self, memodict: Dict[int, object]) -> NoReturn:
         raise TypeError("Bot objects cannot be deepcopied!")
 
     def __eq__(self, other: object) -> bool:
+        """Defines equality condition for the :class:`telegram.Bot` object.
+        Two objects of this class are considered to be equal if their attributes
+        :attr:`bot` are equal.
+
+        Returns:
+            :obj:`True` if both attributes :attr:`bot` are equal. :obj:`False` otherwise.
+        """
```

Comment:
https://github.com/python-telegram-bot/python-telegram-bot/pull/3929#pullrequestreview-1704967873

Would you like to address this in this pr? if not, please open an issue and put it on the next milestone so that we don't forget about it

Reply:
@Bibo-Joshi https://github.com/python-telegram-bot/python-telegram-bot/pull/3929#issuecomment-1793701253 seems breaking to me, it equates Bot to ExtBot (via `==`) as it checks the .bot argument and both objects have same .bot arguments, but hashes are different...?

Reply:
Good obsveration! I see that `__class__` is also part of `TelegramObject.__hash__`, but I guess there are no relevant cases for other classes where one would compage parent class objects with child class objects.
In the case of `Bot`, one could replace `self.__class__` with `Bot` in both `__eq__` and `__hash__` - that way `Bot` and `ExtBot` would be considered "the same", which I would see as expected behavior. Looking at `__hash__`, I also note that it would be cleaner to also relay to `super()` in case `self._bot_user is None`
I still see changes to this as non-breaking as we haven't specified the behavior on this level so far.